aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/sigv4/AppSyncSigV4SignerInterceptor.java

```diff
@@ -61,6 +63,7 @@ public AppSyncSigV4SignerInterceptor(APIKeyAuthProvider apiKey, final String aws
         this.credentialsProvider = null;
         this.cognitoUserPoolsAuthProvider = null;
         this.oidcAuthProvider = null;
+        this.awsLambdaAuthProvider = null;
```

Comment:
Why do we set these providers to null in constructor? Wouldn't they be null anyways?

Reply:
We have to set them to null in each constructor because the variables are `final`.  I don't think this class is designed super well, but refactoring it isn't really something we want to tackle now either.